plugins/pasteurl/trumbowyg.pasteurl.js

```diff
@@ -0,0 +1,61 @@
+/* ===========================================================
+ * trumbowyg.pasteurl.js v1.0
+ * Url paste handling with noembed. Plugin for Trumbowyg
+ * http://alex-d.github.com/Trumbowyg
+ * ===========================================================
+ * Author : Max Seelig
+ *          Facebook : https://facebook.com/maxse
+ *          Website : https://www.maxmade.nl/
+ */
+
+(function ($) {
+    'use strict';
+
+    $.extend(true, $.trumbowyg, {
+        plugins: {
+            pasteImage: {
+                init: function (trumbowyg) {
+                    trumbowyg.pasteHandlers.push(function (pasteEvent) {
+                        try {
+                          var clipboardData = (pasteEvent.originalEvent || pasteEvent).clipboardData;
+                          var pastedData = clipboardData.getData('Text');
+                          var request = null;
+                              if(pastedData.startsWith("http")){
+                                pasteEvent.stopPropagation();
+                                pasteEvent.preventDefault();
+
+                                var url = pastedData;
+                                var query = { url : url.trim() };
+
+                                if (request && request.transport) request.transport.abort();
+
+                                request = $.ajax({
+                                  crossOrigin: true,
+                                  url: "https://noembed.com/embed?nowrap=on",
+                                  type: "GET",
+                                  data: query,
+                                  cache: false,
+                                  dataType: "json",
+                                  success: function(res) {
+                                    if (res.html) {
+                                      let content = res.html;
+                                      trumbowyg.execCmd('insertHTML', content);
```

Comment:
Put this line only one time after the if/else blocks

Reply:
How?
trumbowyg.execCmd('insertHTML', content);  works if content = res.html
trumbowyg.execCmd('insertHTML', content); returns object when content = $('<a>', {href: pastedData, text: pastedData});
trumbowyg.execCmd('insertHTML', content); is also needed in the second error. Can't put it outside of the request because that'll make it lose scope.